src/hocon_cli.erl

```diff
@@ -18,39 +18,280 @@
+    DestinationVMArgs = filename:join(AbsPath, DestinationVMArgsFilename),
+    ?LOG:debug("Generating config in: ~p", [Destination]),
+
+    Schema = load_schema(ParsedArgs),
+    Conf = load_conf(ParsedArgs),
+    LogFun = case proplists:get_value(verbose_env, ParsedArgs) of
+                 true ->
+                     fun(Key, Value) -> ?STDOUT("~s = ~p", [string:join(Key, "."), Value]) end;
+                 false ->
+                     fun(_, _) -> ok end
+             end,
+
+    case hocon_schema:generate(Schema, Conf) of
+        %{error, _X} ->
+            % @TODO print error
+        %    error;
+        NewConfig ->
+            AppConfig = proplists:delete(vm_args, NewConfig),
+            VmArgs = stringify(proplists:get_value(vm_args, NewConfig)),
+
+            %% Prune excess files
+            MaxHistory = proplists:get_value(max_history, ParsedArgs, 3) - 1,
+            prune(Destination, MaxHistory),
+            prune(DestinationVMArgs, MaxHistory),
+
+            case { file:write_file(Destination, io_lib:fwrite("~p.\n", [AppConfig])),
+                file:write_file(DestinationVMArgs, string:join(VmArgs, "\n"))} of
+                {ok, ok} ->
+                    {Destination, DestinationVMArgs};
+                {Err1, Err2} ->
+                    maybe_log_file_error(Destination, Err1),
+                    maybe_log_file_error(DestinationVMArgs, Err2),
+                    error
+            end
+
+    end.
+
+-spec prune(file:name_all(), integer()) -> ok.
+prune(Filename, MaxHistory) ->
+    %% A Filename comes in /Abs/Path/To/something.YYYY.MM.DD.HH.mm.SS.ext
+    %% We want `ls /Abs/Path/To/something.*.ext and delete all but the most
+    %% recent MaxHistory
+    Path = filename:dirname(Filename),
+    Ext = filename:extension(Filename),
+    Base = hd(string:tokens(filename:basename(Filename, Ext), ".")),
+    Files =
+        lists:sort(filelib:wildcard(Base ++ ".*" ++ Ext, Path)),
+
+    delete([ filename:join([Path, F]) || F <- Files], MaxHistory),
+    ok.
+
+-spec delete(file:name_all(), integer()) -> ok.
+delete(Files, MaxHistory) when length(Files) =< MaxHistory ->
+    ok;
+delete([File | Files], MaxHistory) ->
+    case file:delete(File) of
+        ok -> ok;
+        {error, Reason} ->
+            ?LOG:error("Could not delete ~s, ~p", [File, Reason])
+    end,
+    delete(Files, MaxHistory).
+
+-spec maybe_log_file_error(
+    file:filename(), ok |
+    {error, file:posix()  %% copied from file:format_error/1
+    | badarg
+    | terminated
+    | system_limit
+    | { integer(), module(), term() }}) -> ok.
+maybe_log_file_error(_, ok) ->
+    ok;
+maybe_log_file_error(Filename, {error, Reason}) ->
+    ?LOG:error("Error writing ~s: ~s", [Filename, file:format_error(Reason)]),
+    ok.
+
+filename_maker(Filename, Date, Extension) ->
+    {{Y, M, D}, {HH, MM, SS}} = Date,
+    _DestinationFilename =
+        io_lib:format("~s.~p.~s.~s.~s.~s.~s.~s",
```

Comment:
The result of `io_lib:format` typically needs to be wrapped in `lists:flatten`, because it returns an iolist